test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -501,17 +485,25 @@ def set_vntype(self, typestring='myvntype'):
         """
         pbs_home = self.server.pbs_conf['PBS_HOME']
         vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
+        self.logger.info("Setting vntype to %s in %s" %
+                         (typestring, vntype_file))
+        cmd = ['/bin/echo', '%s' % typestring]
+        self.logger.info("Command is: %s" % (cmd))
         with open(vntype_file, "w") as fd:
-            fd.write(typestring)
+            ret = self.du.run_cmd(self.mom.hostname, cmd, stdout=fd, sudo=True)
+        if ret['rc'] != 0:
+            self.skipTest("pbs_cgroups_hook: need root privileges")
```

Comment:
you need to use mkstemp to create the file as root and then rename it using run_copy function (make sure run both as root, uid=0 and gid=0)

Reply:
Was wondering about that. Thanks for the advice. Updated.

Reply:
Updated.